kfs-core/src/main/java/edu/arizona/kfs/gl/businessobject/GecEntryRelationship.java

```diff
@@ -0,0 +1,176 @@
+import java.util.HashMap;
+import java.util.Map;
+
+import org.kuali.kfs.gl.businessobject.Entry;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.krad.bo.PersistableBusinessObjectBase;
+import org.kuali.rice.krad.service.BusinessObjectService;
+
+import edu.arizona.kfs.sys.KFSPropertyConstants;
+
+
+public class GecEntryRelationship extends PersistableBusinessObjectBase {
+    private static final long serialVersionUID = 1107325740786032980L;
+    private static final String TO_STRING_FORMAT = "GecEntryRelationship(enrtryId: %d, gecFdocNumber: %s, gecDocRouteStatusCode: %s, fdocLineTypeCode: %s, acctLineSeqNum: %d)";
+
+    private Long entryId;
+    private String gecDocumentNumber;
+    private Integer gecAcctLineSeqNumber;
+    private String gecDocRouteStatus;
+    private String gecFdocLineTypeCode;
+    private Entry entry;
+
+
+    public GecEntryRelationship() {
+        // OJB requires an empty constructor
+    }
+
+
+    public GecEntryRelationship(Long entryId, String gecDocumentNumber, Integer gecAcctLineSeqNumber, String gecFdocLineTypeCode, String gecDocumentStatus) {
+        this.entryId = entryId;
+        this.gecDocumentNumber = gecDocumentNumber;
+        this.gecAcctLineSeqNumber = gecAcctLineSeqNumber;
+        this.gecFdocLineTypeCode = gecFdocLineTypeCode;
+        this.gecDocRouteStatus = gecDocumentStatus;
+    }
+
+
+    public GecEntryRelationship(Long entryId, String gecDocumentNumber, Integer gecAcctLineSeqNumber, String gecFdocLineTypeCode, String gecDocumentStatus, Entry entry) {
+        this(entryId, gecDocumentNumber, gecAcctLineSeqNumber, gecFdocLineTypeCode, gecDocumentStatus);
+        this.entry = entry;
+    }
+
+
+    public Long getEntryId() {
+        return entryId;
+    }
+
+
+    public void setEntryId(Long entryId) {
+        this.entryId = entryId;
+    }
+
+
+    public String getGecDocumentNumber() {
+        return gecDocumentNumber;
+    }
+
+
+    public void setGecDocumentNumber(String gecDocumentNumber) {
+        this.gecDocumentNumber = gecDocumentNumber;
+    }
+
+
+    public Integer getGecAcctLineSeqNumber() {
+        return gecAcctLineSeqNumber;
+    }
+
+
+    public void setGecAcctLineSeqNumber(Integer gecAcctLineSeqNumber) {
+        this.gecAcctLineSeqNumber = gecAcctLineSeqNumber;
+    }
+
+
+    public String getGecDocRouteStatus() {
+        return gecDocRouteStatus;
+    }
+
+
+    public void setGecDocRouteStatus(String gecDocRouteStatus) {
+        this.gecDocRouteStatus = gecDocRouteStatus;
+    }
+
+
+    public void setEntry(Entry entry) {
+        this.entry = entry;
+    }
+
+
+    public Entry getEntry() {
+        if (entry == null && getEntryId() != null) {
+            Map<String, Object> queryCriteria = new HashMap<String, Object>();
+            queryCriteria.put(KFSPropertyConstants.ENTRY_ID, getEntryId());
+            Collection tmpEntries = SpringContext.getBean(BusinessObjectService.class).findMatching(Entry.class, queryCriteria);
```

Comment:
Oh, I should also mention, the SQL submitted for this column does set this column as UNIQUE too. So basically, we are pretending to be a PK, but without triggering the Rice idiosyncrasies that might get us in trouble with the other PKs that this table has going on.